code/modules/mob/living/carbon/alien/diona/diona_powers.dm

```diff
@@ -79,96 +70,92 @@
 	set category = "Abilities"
 	set name = "Absorb Nymph"
 	set desc = "Absorb a diona nymph into yourself, you will remain in control and gain any biomass it has absorbed."
-	var/list/choices = list()
-	for(var/mob/living/carbon/alien/diona/C in view(1,src))
 
-		if((!(src.Adjacent(C)) || C.gestalt || C == src)) continue//cant steal nymphs right out of other gestalts
-		choices.Add(C)
+	var/list/choices = list()
+	for(var/mob/living/carbon/alien/diona/C in view(1, src))
+		if(!Adjacent(C) || C.gestalt || C == src) //cant steal nymphs right out of other gestalts
+			continue
+		choices += C
 
-	var/mob/living/carbon/alien/diona/M = input(src,"Which nymph do you wish to absorb?") in null|choices
+	var/mob/living/carbon/alien/diona/M = input(src, "Which nymph do you wish to absorb?") in null|choices
 
 	if(!M)
-		to_chat(src, span("warning", "There is nothing nearby to absorb"))
+		to_chat(src, span("warning", "There are no nymphs in your vicinity."))
 	else if(!do_absorb(M))
 		to_chat(src, span("warning", "You fail to merge with \the [M]..."))
 
 
 
 /mob/living/carbon/proc/do_absorb(var/mob/living/carbon/alien/diona/D)
-	if (D.key)
+	if(D.key)
 		//Code for requesting permission goes here. We will return if its denied or ignored
-		to_chat(src, span("warning", "Requesting consent from [D]"))
-		var/r = alert(D,"[src] wishes to absorb your being, and make you a part of their gestalt. If you accept you will join with them, and give up control to be a part of their collective. \nYou will be part of their larger gestalt if they grow later, too, and all of your stored biomass will be transferred to them. You can split away at anytime, but you cannot reclaim the biomass. Do you wish to be absorbed?", "[src] wishes to absorb you", "Yes, we will join!", "No, i wish to remain alone")
-		if (r != "Yes, we will join!")
+		to_chat(src, span("warning", "Requesting consent from [D]."))
+		var/r = alert(D,"[src] wishes to absorb your being, and make you a part of their gestalt. If you accept you will join with them, and give up control to be a part of their collective. \nYou will be part of their larger gestalt if they grow later, too, and all of your stored biomass will be transferred to them. You can split away at anytime, but you cannot reclaim the biomass. Do you wish to be absorbed?", "[src] wishes to absorb you", "Yes, I will join!", "No, I wish to remain alone!")
+		if(r != "Yes, I will join!")
 			to_chat(src, span("warning", "[D] has refused to join you!"))
 			return
-		else
-			if (!(src.Adjacent(D)) || !(istype(D.loc, /turf)))
-				to_chat(src, span("warning", "Something went wrong while trying to absorb [D], cancelling."))
-				return
+		else if(!Adjacent(D) || !isturf(D.loc))
+			to_chat(src, span("warning", "Something went wrong while trying to absorb [D], cancelling."))
```

Comment:
The way this is written feels more like a debug message than an IC message to the Diona. Maybe something like "\The [D] must remain close to the collective if \he wishes to be part of us."